packages/flame/lib/src/sprite_batch.dart

```diff
@@ -144,6 +148,12 @@ class SpriteBatch {
 
   FlippedAtlasStatus _flippedAtlasStatus = FlippedAtlasStatus.none;
 
+  /// A map to keep track of the index of each batch item by its id.
+  final Map<String, int> _idToIndex = {};
+
+  /// Returns all ids currently in the batch (excluding nulls).
+  Iterable<String> get ids => _batchItems.map((item) => item.id!);
```

Comment:
This would currently crash if the ids are null (like you did check for before).

Since you already have the ids in `_idToIndex` you can use the keys from that.
```suggestion
  Iterable<String> get ids => _idToIndex.keys;
```

Another way would be the following (but not necessary here):
```suggestion
  Iterable<String> get ids => [for (final item in _batchItems) ?item.id];
```